### ISSUE TITLE
entityStoragePropertyAssignment false positive on generated proxy classes

### ISSUE THREAD

**Issue by vever001** [mglaman/phpstan-drupal#1012]
# Bug report

`drupal.entityStoragePropertyAssignment` is reported on code that stores no entity storage, whenever `$container->get()` is called with a variable instead of a literal service id.

Every proxy class generated by core's `generate-proxy-class.php` contains such a call, so any project that commits proxies for its lazy services (the documented core workflow) gets this error after updating to 2.1.0.

Root cause: `ContainerDynamicReturnTypeExtension` builds the return type by looping over the constant strings of the argument. A variable has none, so the loop adds nothing and `TypeCombinator::union()` is called with zero types, which produces `never`. Since `never` is a subtype of everything, the rule sees it as entity storage. Same mechanism as #983, which fixed the literal `null` case in the rule but not this path in the extension. Fixing it in the extension covers every rule that consumes the type:

```php
if ($types === []) {
    return $returnType;
}
return TypeCombinator::union(...$types);
```

Seen on mglaman/phpstan-drupal 2.1.0, phpstan 2.2.5, Drupal 11, PHP 8.3.

### Code snippet that reproduces the problem

Generated by `php core/scripts/generate-proxy-class.php`:

```php
protected function lazyLoadItself()
{
    if (!isset($this->service)) {
        $this->service = $this->container->get($this->drupalProxyOriginalServiceId);
    }

    return $this->service;
}
```

### Expected output

No error. Actual output:

```
Storing entity storage as a class property is not recommended. Call
Drupal\Core\Entity\EntityTypeManagerInterface::getStorage() at the call-site instead.
🪪  drupal.entityStoragePropertyAssignment
```
